frontend/pages/campaigns/[campaignId].tsx

```diff
@@ -0,0 +1,102 @@
+import { useRouter } from 'next/router';
+import React from 'react';
+import { canBecomeInteger } from '../../utils/numbers';
+import CampaignCharityCard from '../../components/charities/CampaignCharityCard';
+import { Box, Button, Divider, Stack, Typography } from '@mui/material';
+import { theme } from '../../utils/theme';
+import CircularProgressWithLabel from '../../components/CircularProgressWithLabel';
+import SwiperWrapper from '../../components/swiper/SwiperWrapper';
+import CampaignCharityList from '../../components/charities/CampaignCharityList';
+import api from '../../frontendApis';
+import { GetServerSidePropsContext } from 'next';
+import { CampaignPublicData } from '../../types/campaigns';
+import { StatusMessage } from '../../types/api';
+import { Nullable } from '../../types/utils';
+import { useSnackbar } from 'notistack';
+import { enqueueGCSnackbar } from '../../utils/snackbar';
+import CampaignDescription from '../../components/campaigns/CampaignDescription';
+
+export async function getServerSideProps(
+  context: GetServerSidePropsContext,
+): Promise<{ props: { campaign: Nullable<CampaignPublicData>; message?: Nullable<StatusMessage> } }> {
+  const { campaignId: campaignIdParam } = context.query;
+
+  const campaignId = canBecomeInteger(campaignIdParam) ? Number(campaignIdParam) : undefined;
+
+  if (!campaignId) {
+    return {
+      props: {
+        campaign: null,
+      },
+    };
+  }
+
+  const response = await api.campaigns.getCampaign(campaignId);
+
+  return {
+    props: {
+      campaign: response.payload,
+      message: response.message,
+    },
+  };
+}
+
+type Props = {
+  campaign: Nullable<CampaignPublicData>;
+  message: Nullable<StatusMessage>;
+};
+
+export default function CampaignDetail({ campaign, message }: Props) {
+  const { enqueueSnackbar } = useSnackbar();
+
+  enqueueGCSnackbar(enqueueSnackbar, message);
+
+  if (!campaign) {
+    return null;
+  }
+
+  const numTotalCoupons = campaign.promisedAmount / campaign.couponDenomination;
+  const numCouponsRedeemed = campaign.donations.primaryDonor.amount / campaign.couponDenomination;
+
+  return (
+    <Box sx={{ padding: theme.spacing(2) }}>
```

Comment:
Can we use a container instead? Right now it fills the entire screen and is a little too wide
<img width="1425" alt="Screenshot 2022-10-16 at 1 14 54 PM" src="https://user-images.githubusercontent.com/71375383/196019307-ef253a38-0660-41ac-b932-da2c5301fc13.png">

The quotes are almost covered by the scrollbar

<img width="1438" alt="Screenshot 2022-10-16 at 1 15 19 PM" src="https://user-images.githubusercontent.com/71375383/196019340-e891f371-36e9-4693-81e2-5668101bd0f1.png">

Reply:
Fixed with the Container.
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/75519136/196025531-8edde793-756a-446d-8c7a-c63b9ab34b59.png">